python/ray/serve/deployment.py

```diff
@@ -120,6 +120,17 @@ def __init__(
         self._replica_config = replica_config
         self._docs_path = docs_path
 
+    def _validate_name(self, name: str):
+        if not isinstance(name, str):
+            raise TypeError("name must be a string.")
+
+        # name does not contain #
+        if "#" in name:
+            warnings.warn(
+                f"Deployment names should not contain the '#' character, this will raise an error after 3 releases. "
```

Comment:
Users won't know which release this warning started popping up in.
```suggestion
                f"Deployment names should not contain the '#' character, this will raise an error starting in Ray 2.46.0. "
```